src/main/java/me/botsko/prism/actions/BlockAction.java

```diff
@@ -30,574 +30,534 @@
+                    final SignActionData signActionData = new SignActionData();
+                    final Sign sign = (Sign) state;
+                    signActionData.lines = sign.getLines();
+                    actionData = signActionData;
+                }
+        }
+    }
+
+    private void setBlockRotation(BlockState block, SkullActionData skullActionData) {
+        if (block.getBlockData() instanceof Rotatable) {
+            final Rotatable r = (Rotatable) block.getBlockData();
+            skullActionData.rotation = r.getRotation().toString();
+        } else {
+            final Directional d = (Directional) block.getBlockData();
+            skullActionData.rotation = d.getFacing().name().toLowerCase();
+        }
+    }
+
+
+    @Override
+    public boolean hasExtraData() {
+        return actionData != null;
+    }
+
+    @Override
+    public String serialize() {
+        return gson().toJson(actionData);
+    }
+
+    @Override
+    public void deserialize(String data) {
+        if (data != null && data.startsWith("{")) {
+            if (getMaterial() == Material.PLAYER_HEAD || getMaterial() == Material.PLAYER_WALL_HEAD) {
+                actionData = gson().fromJson(data, SkullActionData.class);
+            } else if (getMaterial() == Material.SPAWNER) {
+                actionData = gson().fromJson(data, SpawnerActionData.class);
+            } else if (Tag.SIGNS.isTagged(getMaterial())) {
+                actionData = gson().fromJson(data, SignActionData.class);
+            } else if (getMaterial() == Material.COMMAND_BLOCK) {
+                actionData = new CommandActionData();
+                ((CommandActionData) actionData).command = data;
+            }
+        }
+    }
+
+    private BlockActionData getActionData() {
+        return actionData;
+    }
+
+
+    @Override
+    public String getNiceName() {
+        String name = "";
+        BlockActionData blockActionData = getActionData();
+
+        if (blockActionData instanceof SkullActionData) {
+            final SkullActionData ad = (SkullActionData) blockActionData;
+            name += ad.skullType + " ";
+        } else if (blockActionData instanceof SpawnerActionData) {
+            final SpawnerActionData ad = (SpawnerActionData) blockActionData;
+            name += ad.entityType + " ";
+        }
+        name += Prism.getItems().getAlias(getMaterial(), getBlockData());
+        if (blockActionData instanceof SignActionData) {
+            final SignActionData ad = (SignActionData) blockActionData;
+            if (ad.lines != null && ad.lines.length > 0) {
+                name += " (" + TypeUtils.join(ad.lines, ", ") + ")";
+            }
+        } else if (blockActionData instanceof CommandActionData) {
+            final CommandActionData ad = (CommandActionData) blockActionData;
+            name += " (" + ad.command + ")";
+        }
+        if (getActionType().getName().equals("crop-trample") && getMaterial() == Material.AIR) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Unnecessary use of fully qualified name 'Material.AIR' due to existing static import 'org.bukkit.Material.*'](https://app.codacy.com/gh/AddstarMC/Prism-Bukkit/pullRequest?prid=5446892)